window/src/os/windows/window.rs

```diff
@@ -1143,10 +1160,30 @@ unsafe fn mouse_move(hwnd: HWND, _msg: UINT, wparam: WPARAM, lparam: LPARAM) ->
             modifiers,
         };
 
-        inner
-            .borrow_mut()
-            .events
-            .dispatch(WindowEvent::MouseEvent(event));
+        inner.events.dispatch(WindowEvent::MouseEvent(event));
+        Some(0)
+    } else {
+        None
+    }
+}
+
+unsafe fn mouse_leave(hwnd: HWND, _msg: UINT, _wparam: WPARAM, _lparam: LPARAM) -> Option<LRESULT> {
+    if let Some(inner) = rc_from_hwnd(hwnd) {
+        let mut inner = inner.borrow_mut();
+
+        inner.track_mouse_leave = false;
+
+        let coords = Point::new(-1, -1);
```

Comment:
Could we get the actual mouse screen coordinates and compute the effective client coordinates here instead?

I'm thinking about future features such as dragging and dropping tabs and I think it might be surprising if this event interleaved and made it appear as though the mouse suddenly moved off screen.

Reply:
I thought about that too, so I did some testing. Turns out the mouse leave event is not fired when the mouse leaves the window while captured, and it only fires after all mouse events are done, so it's pretty safe. I did that `-1` as a hacky way to signal that the mouse is out of the window, but I think it would be better if there's a specific API for that.

Reply:
I think using [GetCursorPos](https://docs.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-getcursorpos) to get the screen coordinates and then translating those to window coordinates feels more robust overall

Reply:
But then again, that would imply Windows-specific code. An API to signal mouse out of the window, which is independent of mouse position, would likely be useful to all supported OSes. Maybe a `WindowEvent::LeaveWindow` or something similar, as opposed to a `MouseEvent`, because in this case all other mouse related stuff are irrelevant.

Reply:
I've implemented my suggested API and used it to fix the bug on Windows, Wayland and X11.